src/utils/setjsApi.ts

```diff
@@ -1,9 +1,27 @@
 import { provider } from 'web3-core'
 import Set from "set.js"
-import { StreamingFeeInfo } from "set.js/dist/types/src/types";
+import { SetDetails, StreamingFeeInfo } from "set.js/dist/types/src/types";
 
 import { basicIssuanceModuleAddress, controllerAddress, debtIssuanceModuleAddress, navIssuanceModuleAddress, protocolViewerAddress, streamingFeeModuleAddress, setTokenCreatorAddress, tradeModuleAddress, masterOracleAddress, governanceModuleAddress } from "constants/ethContractAddresses";
 
+export async function getTokenSupply(web3Provider: provider, productAddresses: string[]): Promise<SetDetails[]> {
```

Comment:
This is interesting. Given we're fetching more than just Set Details and not just Token Supply here, does it make sense to rename this?

Or maybe it could make sense to alter `TokenSupplyProvider` to be a general `SetDetailsProvider`? And add a selector for the `tokenSupply` part of the `setDetails` response.

If we want to do that in a follow up PR instead of here that's fine too.

Reply:
I'm down to do this rename in a future PR when I tackle reading product components via set.js